src/platform/audio/alaudiomanager.cpp

```diff
@@ -0,0 +1,1587 @@
+        }
+    }
+
+    if (m_volumeSet) {
+        m_volumeSet = false;
+    }
+}
+
+/**
+ * Processes the fading audio lists.
+ */
+void ALAudioManager::Process_Fading_List()
+{
+    for (auto it = m_fadingList.begin(); it != m_fadingList.end();) {
+        if (*it == nullptr) {
+            // TODO BUGFIX This appears to be an infinite loop if any entry is null.
+            continue;
+        }
+
+        if ((*it)->openal.time_fading < Get_Audio_Settings()->Get_Time_To_Fade()) {
+            ++(*it)->openal.time_fading;
+            float adjustment = (float)(*it)->openal.time_fading / (float)Get_Audio_Settings()->Get_Time_To_Fade();
+            float effective_vol = (float)(1.0f - adjustment) * Get_Effective_Volume((*it)->openal.audio_event);
+            alSourcef((*it)->openal.source, AL_GAIN, effective_vol);
+            ++it;
+        } else {
+            (*it)->openal.stopped = 1;
+            (*it)->openal.disable_loops = true;
+            Release_Playing_Audio(*it);
+            it = m_fadingList.erase(it);
+        }
+    }
+}
+
+/**
+ * Processes the stopped audio lists.
+ */
+void ALAudioManager::Process_Stopped_List()
+{
+    for (auto it = m_stoppedList.begin(); it != m_stoppedList.end(); it = m_stoppedList.erase(it)) {
+        if (*it != nullptr) {
+            Release_Playing_Audio(*it);
+        }
+    }
+}
+
+/**
+ * Stops the playing audio sample.
+ */
+void ALAudioManager::Release_Playing_Audio(PlayingAudio *audio)
+{
+    if (audio->openal.audio_event->Get_Event_Info()->Get_Event_Type() == EVENT_SOUND) {
+        if (audio->openal.playing_type != 0) {
+            m_soundManager->Notify_Of_3D_Sample_Completion();
+        } else {
+            m_soundManager->Notify_Of_2D_Sample_Completion();
+        }
+    }
+
+    if (audio->openal.buffer)
+        alDeleteBuffers(1, &audio->openal.buffer);
+    alDeleteSources(1, &audio->openal.source);
+    Close_File(audio->openal.file_handle);
+
+    if (audio->openal.release_event) {
+        Release_Audio_Event_RTS(audio->openal.audio_event);
+    }
+
+    delete audio;
+}
+
+/**
+ * Releases all playing audio samples.
+ */
+void ALAudioManager::Stop_All_Audio_Immediately()
+{
+    for (auto it = m_globalAudioList.begin(); it != m_globalAudioList.end(); it = m_globalAudioList.erase(it)) {
+        if (*it != nullptr) {
+            Release_Playing_Audio(*it);
+        }
+    }
+
+    for (auto it = m_positionalAudioList.begin(); it != m_positionalAudioList.end(); it = m_positionalAudioList.erase(it)) {
+        if (*it != nullptr) {
+            Release_Playing_Audio(*it);
+        }
+    }
+
+    for (auto it = m_streamList.begin(); it != m_streamList.end(); it = m_streamList.erase(it)) {
+        if (*it != nullptr) {
+            Release_Playing_Audio(*it);
+        }
+    }
+
+    for (auto it = m_fadingList.begin(); it != m_fadingList.end(); it = m_fadingList.erase(it)) {
+        if (*it != nullptr) {
+            Release_Playing_Audio(*it);
+        }
+    }
+}
```

Comment:
What about `m_quickAudioList` ?

Reply:
It seems to be used for force_Play, which i did not implement yet